dotCMS/src/main/java/com/dotcms/security/secret/SecretsStoreKeyStoreImpl.java

```diff
@@ -0,0 +1,303 @@
+ *
+ */
+public class SecretsStoreKeyStoreImpl implements SecretsStore {
+
+    private static final String SECRETS_STORE_FILE = "dotSecretsStore.p12";
+    private static final String SECRETS_STORE_KEYSTORE_TYPE = "pkcs12";
+    private static final String SECRETS_STORE_SECRET_KEY_FACTORY_TYPE = "PBE";
+    protected static final String SECRETS_CACHE_GROUP = "SECRETS_CACHE";
+    private static final String SECRETS_KEYSTORE_PASSWORD_KEY = "SECRETS_KEYSTORE_PASSWORD_KEY";
+    private static final String SECRETS_KEYSTORE_FILE_PATH_KEY = "SECRETS_KEYSTORE_FILE_PATH_KEY";
+    private final String secretsKeyStorePath;
+
+    private static String getSecretStorePath() {
+        final Supplier<String> supplier = () -> APILocator.getFileAssetAPI().getRealAssetsRootPath()
+                + File.separator + "server" + File.separator + SECRETS_STORE_FILE;
+        return Config.getStringProperty(SECRETS_KEYSTORE_FILE_PATH_KEY, supplier.get());
+    }
+
+    @VisibleForTesting
+    protected SecretsStoreKeyStoreImpl(final String secretsKeyStorePath) {
+        this.secretsKeyStorePath = secretsKeyStorePath;
+    }
+
+    protected SecretsStoreKeyStoreImpl() {
+        this(getSecretStorePath());
+    }
+
+    /**
+     * returns the password to the store - it can be set as a config variable (which can read from an
+     * environment or system property) or will default to the SHA-256 digest of the clusterId
+     * 
+     * @return
+     */
+    @VisibleForTesting
+    private char[] loadStorePassword() {
+        return getFromCache(SECRETS_KEYSTORE_PASSWORD_KEY,
+                () -> Config.getStringProperty(SECRETS_KEYSTORE_PASSWORD_KEY, digest(ClusterFactory.getClusterId()))
+                                .toCharArray());
+
+    }
+
+    /**
+     * This will create a Keystore file for reading/writing if there is not one already there,
+     * otherwise, it will return the one there
+     * 
+     * @return
+     */
+    private File createStoreIfNeeded() {
+        final File secretStoreFile = new File(secretsKeyStorePath);
+        if (!secretStoreFile.exists()) {
+            try {
+                final KeyStore keyStore = KeyStore.getInstance(SECRETS_STORE_KEYSTORE_TYPE);
+                keyStore.load(null, loadStorePassword());
+                flushCache();
+                saveSecretsStore(keyStore);
+            } catch (Exception e) {
+                Logger.warn(this.getClass(), "unable to create secrets store " + SECRETS_STORE_FILE + ": " + e);
+                throw new DotRuntimeException(e);
+            }
+        }
+        return secretStoreFile;
+    }
+
+
+    /**
+     * loads up the Keystore from disk
+     * 
+     * @return
+     */
+    private KeyStore getSecretsStore() {
+        final File secretStoreFile = createStoreIfNeeded();
+        try {
+            final KeyStore keyStore = KeyStore.getInstance(SECRETS_STORE_KEYSTORE_TYPE);
+            try (InputStream in = Files.newInputStream(secretStoreFile.toPath())) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid variables with short names like in](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4623512)